brave/brave-core
# Refactor SetContributionAutoInclude to use Callback

Refactor function and affected areas to use callback`OnSetContributionAutoInclude`

Resolves brave/brave-browser#4642

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on

## Test Plan:

    1. Clean profile
    2. Join rewards
    3. Browse to a site and view long enough to add to auto contribution table (approx 8 seconds).
    4. Open brave://rewards 
    5. Scroll to auto contribution and exclude site from list, entry should disappear
    6. Restore publisher by clicking "show all", clicking "excluded sites" tab and clicking "restore".
    7. Repeat step 3 with same site
    8. Site should once again show up in auto contribution table. 


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
